Tasks/AzureRmWebAppDeployment/task.json

```diff
@@ -58,12 +68,77 @@
             "helpMarkDown": "Enter or Select the name of an existing Azure App Service."
         },
         {
+            "name": "ServerfarmId",
+            "type": "pickList",
+            "label": "Serverfarm id",
+            "defaultValue": "",
+            "required": false,
+            "visibleRule": "WebAppName = invalidwebappname",
+            "properties": {
+                "EditableOptions": "false"
+            }
+        },
+        {
+            "name": "WebAppKind",
+            "type": "pickList",
+            "label": "Web app kind",
+            "defaultValue": "",
+            "required": false,
+            "visibleRule": "WebAppName = invalidwebappname",
+            "properties": {
+                "EditableOptions": "false"
+            }
+        },   
+        {
+            "name": "DockerNamespace",
+            "type": "string",
+            "label": "Registry\\Namespace",
+            "defaultValue": "",
+            "required": true,
+            "visibleRule": "WebAppKind = linux",
+            "helpMarkDown": "Docker repository namespace."
+        },
+        {
+            "name": "DockerRepository",
+            "type": "string",
+            "label": "Repository",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": "false"
```

Comment:
string type has impact of properties. Forgot to remove while changing from pickList to string.